spec/unit/type/file/content_spec.rb

```diff
@@ -227,6 +227,30 @@
           end
         end
       end
+
+      SAVED_TIME = Time.now
+      [:ctime, :mtime].each do |time_stat|
+        [["older", SAVED_TIME-1, false], ["same", SAVED_TIME, true], ["newer", SAVED_TIME+1, true]].each do
+          |compare, target_time, success|
```

Comment:
Directories and symlinks specifically don't cause an issue. However, testing that with recursive directory sourcing found another issue with the checksum_type not propagating correctly. I'll have to re-open or start a new subtask.